plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java

```diff
@@ -479,17 +493,27 @@ else if (metadata.getStatisticType().equals(ColumnStatisticType.NUMBER_OF_DISTIN
             return info;
         }
 
+        Map<String, List<Object>> columnValues = getColumnValues(tableName, info, distinctValues, minimums, maximums);
         return info.withColumns(columns.stream().map(column -> {
             if (isCharacterType(column.type()) || !minimums.containsKey(column.name())) {
                 return column;
             }
             FakerColumnHandle handle = column.handle();
             Map<String, Object> properties = new HashMap<>(column.metadata().getProperties());
-            Object min = minimums.get(column.name());
-            Object max = maximums.get(column.name());
-            handle = handle.withDomain(Domain.create(ValueSet.ofRanges(Range.range(column.type(), min, true, max, true)), false));
-            properties.put(MIN_PROPERTY, Literal.format(column.type(), min));
-            properties.put(MAX_PROPERTY, Literal.format(column.type(), max));
+            if (columnValues.containsKey(column.name())) {
+                List<Object> allowedValues = columnValues.get(column.name());
+                handle = handle.withDomain(Domain.create(ValueSet.copyOf(column.type(), allowedValues), false));
+                properties.put(ALLOWED_VALUES_PROPERTY, columnValues.get(column.name()).stream()
+                        .map(value -> Literal.format(column.type(), value))
+                        .collect(toImmutableList()));
```

Comment:
what is property size limit?

Reply:
I don't know, but the max number of values is configurable, with a reasonable default of 1000. This might throw a some exception for big values, but I don't think we have to prevent that.